Extensions/ParticleSystem/particleemitterobject.js

```diff
@@ -1,56 +1,169 @@
 /**
-
-GDevelop - Particle System Extension
-Copyright (c) 2010-2016 Florian Rival (Florian.Rival@gmail.com)
-This project is released under the MIT License.
+ * @fileoverview
+ * @name Particle-System-Extension
+ * @copyright Copyright (c) 2010-2016 Florian Rival (Florian.Rival@gmail.com)
+ * @author Florian Rival
+ * @license MIT
+ * GDevelop - Particle System Extension
+ * Copyright (c) 2010-2016 Florian Rival (Florian.Rival@gmail.com)
+ * This project is released under the MIT License.
 */
 
 
-gdjs.ParticleEmitterObject = function(runtimeScene, objectData){
-    gdjs.RuntimeObject.call(this, runtimeScene, objectData);
-
-    this._renderer = new gdjs.ParticleEmitterObjectRenderer(runtimeScene, this, objectData);
-
-    this.singleAngle = objectData.emissionEditionSimpleMode;
-    this.angleA = objectData.emitterAngleA;
-    this.angleB = objectData.emitterAngleB;
-    this.forceMin = objectData.emitterForceMin;
-    this.forceMax = objectData.emitterForceMax;
-    this.zoneRadius = objectData.zoneRadius;
-    this.lifeTimeMin = objectData.particleLifeTimeMin;
-    this.lifeTimeMax = objectData.particleLifeTimeMax;
-    this.gravityX = objectData.particleGravityX;
-    this.gravityY = objectData.particleGravityY;
-    this.colorR1 = objectData.particleRed1;
-    this.colorR2 = objectData.particleRed2;
-    this.colorG1 = objectData.particleGreen1;
-    this.colorG2 = objectData.particleGreen2;
-    this.colorB1 = objectData.particleBlue1;
-    this.colorB2 = objectData.particleBlue2;
-    this.size1 = objectData.particleSize1;
-    this.size2 = objectData.particleSize2;
-    this.sizeParam = objectData.sizeParam;
-    this.alpha1 = objectData.particleAlpha1;
-    this.alpha2 = objectData.particleAlpha2;
-    this.rendererType = objectData.rendererType;
-    this.rendererParam1 = objectData.rendererParam1;
-    this.rendererParam2 = objectData.rendererParam2;
-    /** @type string */
-    this.texture = objectData.textureParticleName;
-    this.flow = objectData.flow;
-    this.tank = objectData.tank;
-    this.destroyWhenNoParticles = objectData.destroyWhenNoParticles;
+// FIX-ME @4ian correct with real types (thoose are assunmptions only) and add desciptions
+/**
+ * @typedef {Object} particleObjectDataType
```

Comment:
I think we should follow the convention: gdjs[ObjectName]ObjectDataType

So here: `gdjsParticleEmitterObjectDataType`

(fairly long, but the goal is that you don't have to write it apart from here and in the constructor).
Ideally I prefer types to start with a capital letter though... but we can consider gdjs as a namespace.

Reply:
The type can be named however you want as the combination between it and objectData counts in the end, but that combination is meant to be used by JS Events users to autocomplete their objectData. If it is not easily accessible, they wont be able to do it like this:
```js
/** @type {particleObjectData} */
let objectData = {
    ...
}
```
Is that is what you meant

Reply:
If we want to be sure it's easily discoverable, then let's go for `[ObjectName]ObjectDataType`?

Without the gdjs at the beginning

Reply:
If you really care about structures, we can also (later) create interfaces in a namespace like gdjs.types for the users to be used instead of typedefs then you can give whatever name you want to them. For now I would stick to easy to find names.